catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswEndpoint.java

```diff
@@ -670,25 +670,26 @@ private int deleteRecords(DeleteAction deleteAction)
         queryRequest = queryFactory.updateQueryRequestTags(queryRequest,
                 schemaTransformerManager.getTransformerSchemaForId(deleteAction.getTypeName()));
 
-        Iterable<List<Result>> resultLists = Iterables.partition(ResultIterable.resultIterable(
-                framework,
-                queryRequest), DEFAULT_BATCH);
+        ResultIterable resultIterable = ResultIterable.resultIterable(framework, queryRequest);
```

Comment:
❓ To really make sure we're not causing OOM, should we do something like this instead?
```
Iterable<Result> results = Iterables.limit(ResultIterable(
         framework,
         queryRequest), DEFAULT_BATCH);

while (results.hasNext()) {
    Set<String> idsToDelete = results.next().stream()....;

    if (!idsToDelete.isEmpty()) {
        // Delete batch
    }

    results = Iterables.limit(ResultIterable(
         framework,
         queryRequest), DEFAULT_BATCH);
}
```

Reply:
If you wish to limit the amount of results there is a method now to pass a maximum result limit to fetch 
https://github.com/codice/ddf/blob/master/catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ResultIterable.java#L97-L98

Reply:
@lessarderic In regards to the actual comment, I suggested your way as an alternative but @coyotesqrl thought that it would be best to do it wholesale like it is being done here. 

We are only mapping to IDs, which means that they would have to be requesting... a very very large amount of ids in order to overwhelm the system purely with id strings.

Reply:
Yeah, I did think about that. There are two things to consider here imho:

1. Multiple operations could be going on at the same time, so if someone does a large delete and bloats the memory with IDs, that could cause other requested being processed to blow up the memory.
1. We've never tried to see how many IDs can be passed down to the Catalog Framework/Solr before things blow up like they did on the request side.

Reply:
The Max Boolean clauses limit of solr usually blows up (1024 max clauses). (For a single request)